src/liballoc/vec.rs

```diff
@@ -748,28 +748,64 @@ impl<T> Vec<T> {
         self
     }
 
-    /// Sets the length of a vector.
+    /// Forces the length of a vector to a particular value.
     ///
-    /// This will explicitly set the size of the vector, without actually
-    /// modifying its buffers, so it is up to the caller to ensure that the
-    /// vector is actually the specified size.
+    /// This is a low-level operation that maintains none of the normal
+    /// invariants of the type.  Normally changing the length of a `Vec`
+    /// is done using one of the safe operations instead, such as
+    /// [`truncate`], [`resize`], [`extend`], or [`clear`].
     ///
-    /// # Examples
+    /// [`truncate`]: #method.truncate
+    /// [`resize`]: #method.resize
+    /// [`extend`]: #method.extend-1
+    /// [`clear`]: #method.clear
     ///
-    /// ```
-    /// use std::ptr;
+    /// # Safety
     ///
-    /// let mut vec = vec!['r', 'u', 's', 't'];
+    /// - `new_len` must be less than or equal to `capacity()`.
+    /// - All elements between past the previous end up to the `new_len`
```

Comment:
It's hard to parse exactly what  "between past the previous end up to" means, particularly true if you are not a native English speaker. Can we give a backup wording that relies on a more formal and universal language (i.e. stating it in math/Rust-like notation)?

Reply:
> All elements between `Vec::len()` and `new_len` must be initialized by the user after `set_len` returns

seems like a better wording, as it describes *when* the initialization should occur. There’s a slight ambiguity in the wording (both currently PRd and my proposed) that one could interpret

```rust
fn weird_fn<T>(v: &mut Vec<T>) { 
    let old_len = v.len();
    v.set_len(0);
    // ...
    v.set_len(old_len);
}
```

as invalid… would be great to find a wording that does not have such ambiguity.

Reply:
I tried range notation; let me know if that's better.

Reply:
Yeah, much better!